fixed golangci-lint
```
../internal/kms/azure_vault.go:195:22: unnecessary conversion (unconvert)
			config[k] = string(v)
			                  ^
../internal/kms/keyprotect.go:212:22: unnecessary conversion (unconvert)
			config[k] = string(v)
			                  ^
../internal/kms/kmip.go:310:22: unnecessary conversion (unconvert)
			config[k] = string(v)
			                  ^
```

--- a/internal/kms/aws_metadata.go
+++ b/internal/kms/aws_metadata.go
@@ -28,7 +28,6 @@ import (
 	awsCreds "github.com/aws/aws-sdk-go/aws/credentials"
 	awsSession "github.com/aws/aws-sdk-go/aws/session"
 	awsKMS "github.com/aws/aws-sdk-go/service/kms"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
@@ -185,25 +184,17 @@ func (kms *awsMetadataKMS) GetSecret(ctx context.Context, volumeID string) (stri
 }
 
 func (kms *awsMetadataKMS) getSecrets() (map[string]interface{}, error) {
-	c, err := k8s.NewK8sClient()
+	secretData, err := k8s.GetSecret(kms.secretName, kms.namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to Kubernetes to "+
-			"get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
-	}
-
-	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
-		kms.secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to get Secret %s/%s: %w",
-			kms.namespace, kms.secretName, err)
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
 	}
 
 	config := make(map[string]interface{})
 
-	for k, v := range secret.Data {
+	for k, v := range secretData {
 		switch k {
 		case awsSecretAccessKey, awsAccessKey, awsSessionToken, awsCMK:
-			config[k] = string(v)
+			config[k] = v
 		default:
 			return nil, fmt.Errorf("unsupported option for KMS "+
 				"provider %q: %s", kmsTypeAWSMetadata, k)

--- a/internal/kms/aws_sts_metadata.go
+++ b/internal/kms/aws_sts_metadata.go
@@ -30,7 +30,6 @@ import (
 	awsCreds "github.com/aws/aws-sdk-go/aws/credentials"
 	awsSession "github.com/aws/aws-sdk-go/aws/session"
 	awsKMS "github.com/aws/aws-sdk-go/service/kms"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
@@ -160,24 +159,16 @@ func (as *awsSTSMetadataKMS) DecryptDEK(ctx context.Context, _, encryptedDEK str
 
 // getSecrets returns required STS configuration options from the Kubernetes Secret.
 func (as *awsSTSMetadataKMS) getSecrets() (map[string]string, error) {
-	c, err := k8s.NewK8sClient()
+	secretData, err := k8s.GetSecret(as.secretName, as.namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to Kubernetes to "+
-			"get Secret %s/%s: %w", as.namespace, as.secretName, err)
-	}
-
-	secret, err := c.CoreV1().Secrets(as.namespace).Get(context.TODO(),
-		as.secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to get Secret %s/%s: %w",
-			as.namespace, as.secretName, err)
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", as.namespace, as.secretName, err)
 	}
 
 	config := make(map[string]string)
-	for k, v := range secret.Data {
+	for k, v := range secretData {
 		switch k {
 		case awsSTSRoleARNKey, awsSTSRegionKey, awsSTSCMKARNKey:
-			config[k] = string(v)
+			config[k] = v
 		default:
 			return nil, fmt.Errorf("unsupported option for KMS "+
 				"provider %q: %s", kmsTypeAWSMetadata, k)

--- a/internal/kms/azure_vault.go
+++ b/internal/kms/azure_vault.go
@@ -25,7 +25,6 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
@@ -184,23 +183,16 @@ func (kms *azureKMS) getService() (*azsecrets.Client, error) {
 }
 
 func (kms *azureKMS) getSecrets() (map[string]interface{}, error) {
-	c, err := k8s.NewK8sClient()
+	secretData, err := k8s.GetSecret(kms.secretName, kms.namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to kubernetes to "+
-			"get secret %s/%s: %w", kms.namespace, kms.secretName, err)
-	}
-
-	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
-		kms.secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to get secret %s/%s: %w", kms.namespace, kms.secretName, err)
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
 	}
 
 	config := make(map[string]interface{})
-	for k, v := range secret.Data {
+	for k, v := range secretData {
 		switch k {
 		case azureClientCertificate:
-			config[k] = string(v)
+			config[k] = v
 		default:
 			return nil, fmt.Errorf("unsupported option for KMS provider %q: %s", kmsTypeAzure, k)
 		}

--- a/internal/kms/keyprotect.go
+++ b/internal/kms/keyprotect.go
@@ -25,7 +25,6 @@ import (
 	"github.com/ceph/ceph-csi/internal/util/k8s"
 
 	kp "github.com/IBM/keyprotect-go-client"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
@@ -200,25 +199,17 @@ func (kms *keyProtectKMS) GetSecret(ctx context.Context, volumeID string) (strin
 }
 
 func (kms *keyProtectKMS) getSecrets() (map[string]interface{}, error) {
-	c, err := k8s.NewK8sClient()
+	secretData, err := k8s.GetSecret(kms.secretName, kms.namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to Kubernetes to "+
-			"get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
-	}
-
-	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
-		kms.secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to get Secret %s/%s: %w",
-			kms.namespace, kms.secretName, err)
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
 	}
 
 	config := make(map[string]interface{})
 
-	for k, v := range secret.Data {
+	for k, v := range secretData {
 		switch k {
 		case keyProtectServiceAPIKey, KeyProtectCustomerRootKey, keyProtectSessionToken, keyProtectCRK:
-			config[k] = string(v)
+			config[k] = v
 		default:
 			return nil, fmt.Errorf("unsupported option for KMS "+
 				"provider %q: %s", kmsTypeKeyProtectMetadata, k)

--- a/internal/kms/kmip.go
+++ b/internal/kms/kmip.go
@@ -34,7 +34,6 @@ import (
 	"github.com/gemalto/kmip-go/kmip14"
 	"github.com/gemalto/kmip-go/ttlv"
 	"github.com/google/uuid"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
@@ -299,24 +298,16 @@ func (kms *kmipKMS) GetSecret(ctx context.Context, volumeID string) (string, err
 
 // getSecrets returns required options from the Kubernetes Secret.
 func (kms *kmipKMS) getSecrets() (map[string]string, error) {
-	c, err := k8s.NewK8sClient()
+	secretData, err := k8s.GetSecret(kms.secretName, kms.namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to Kubernetes to "+
-			"get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
-	}
-
-	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
-		kms.secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to get Secret %s/%s: %w",
-			kms.namespace, kms.secretName, err)
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
 	}
 
 	config := make(map[string]string)
-	for k, v := range secret.Data {
+	for k, v := range secretData {
 		switch k {
 		case kmipClientKey, kmipCLientCert, kmipCACert, kmipUniqueIdentifier:
-			config[k] = string(v)
+			config[k] = v
 		default:
 			return nil, fmt.Errorf("unsupported option for KMS "+
 				"provider %q: %s", kmsTypeKMIP, k)

--- a/internal/kms/secretskms.go
+++ b/internal/kms/secretskms.go
@@ -29,7 +29,6 @@ import (
 	"github.com/ceph/ceph-csi/internal/util/k8s"
 
 	"golang.org/x/crypto/scrypt"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
@@ -265,26 +264,18 @@ func (kms secretsMetadataKMS) fetchEncryptionPassphrase(
 		secretNamespace = defaultNamespace
 	}
 
-	c, err := k8s.NewK8sClient()
+	secretData, err := k8s.GetSecret(secretName, secretNamespace)
 	if err != nil {
-		return "", fmt.Errorf("can not get Secret %s/%s, failed to "+
-			"connect to Kubernetes: %w", secretNamespace, secretName, err)
+		return "", fmt.Errorf("failed to get Secret %s/%s: %w", secretNamespace, secretName, err)
 	}
 
-	secret, err := c.CoreV1().Secrets(secretNamespace).Get(context.TODO(),
-		secretName, metav1.GetOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to get Secret %s/%s: %w",
-			secretNamespace, secretName, err)
-	}
-
-	passphraseValue, ok := secret.Data[encryptionPassphraseKey]
+	passphraseValue, ok := secretData[encryptionPassphraseKey]
 	if !ok {
 		return "", fmt.Errorf("missing %q in Secret %s/%s",
 			encryptionPassphraseKey, secretNamespace, secretName)
 	}
 
-	return string(passphraseValue), nil
+	return passphraseValue, nil
 }
 
 // generateCipher returns a AEAD cipher based on a passphrase and salt

--- a/internal/kms/vault_sa.go
+++ b/internal/kms/vault_sa.go
@@ -22,6 +22,8 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/ceph/ceph-csi/internal/util/k8s"
+
 	"github.com/libopenstorage/secrets/vault"
 	authenticationv1 "k8s.io/api/authentication/v1"
 	corev1 "k8s.io/api/core/v1"
@@ -304,14 +306,14 @@ func (kms *vaultTenantSA) getToken() (string, error) {
 	}
 
 	for _, secretRef := range sa.Secrets {
-		secret, sErr := c.CoreV1().Secrets(kms.Tenant).Get(context.TODO(), secretRef.Name, metav1.GetOptions{})
-		if sErr != nil {
-			return "", fmt.Errorf("failed to get Secret %s/%s: %w", kms.Tenant, secretRef.Name, sErr)
+		secretData, err := k8s.GetSecret(secretRef.Name, kms.Tenant)
+		if err != nil {
+			return "", fmt.Errorf("failed to get Secret %s/%s: %w", kms.Tenant, secretRef.Name, err)
 		}
 
-		token, ok := secret.Data["token"]
+		token, ok := secretData["token"]
 		if ok {
-			return string(token), nil
+			return token, nil
 		}
 	}
 

--- a/internal/kms/vault_tokens.go
+++ b/internal/kms/vault_tokens.go
@@ -510,41 +510,31 @@ func (vtc *vaultTenantConnection) getK8sClient() (*kubernetes.Clientset, error) 
 }
 
 func (kms *vaultTokensKMS) getToken() (string, error) {
-	c, err := kms.getK8sClient()
+	secretData, err := k8s.GetSecret(kms.TokenName, kms.Tenant)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get Secret %s/%s: %w", kms.Tenant, kms.TokenName, err)
 	}
 
-	secret, err := c.CoreV1().Secrets(kms.Tenant).Get(context.TODO(), kms.TokenName, metav1.GetOptions{})
-	if err != nil {
-		return "", err
-	}
-
-	token, ok := secret.Data[vaultTokenSecretKey]
+	token, ok := secretData[vaultTokenSecretKey]
 	if !ok {
 		return "", errors.New("failed to parse token")
 	}
 
-	return string(token), nil
+	return token, nil
 }
 
 func (vtc *vaultTenantConnection) getCertificate(tenant, secretName, key string) (string, error) {
-	c, err := vtc.getK8sClient()
+	secretData, err := k8s.GetSecret(secretName, tenant)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get Secret %s/%s: %w", tenant, secretName, err)
 	}
 
-	secret, err := c.CoreV1().Secrets(tenant).Get(context.TODO(), secretName, metav1.GetOptions{})
-	if err != nil {
-		return "", err
-	}
-
-	cert, ok := secret.Data[key]
+	cert, ok := secretData[key]
 	if !ok {
 		return "", errors.New("failed to parse certificates")
 	}
 
-	return string(cert), nil
+	return cert, nil
 }
 
 // isTenantConfigOption return true if a tenant may (re)configure the option in

--- a/internal/rbd/rbd_healer.go
+++ b/internal/rbd/rbd_healer.go
@@ -57,18 +57,16 @@ func accessModeStrToInt(mode v1.PersistentVolumeAccessMode) csi.VolumeCapability
 }
 
 // getSecret get the secret details by name.
-func getSecret(c *k8s.Clientset, ns, name string) (map[string]string, error) {
+func getSecret(ns, name string) (map[string]string, error) {
 	deviceSecret := make(map[string]string)
 
-	secret, err := c.CoreV1().Secrets(ns).Get(context.TODO(), name, metav1.GetOptions{})
+	secretData, err := kubeclient.GetSecret(name, ns)
 	if err != nil {
-		log.ErrorLogMsg("get secret failed, err: %v", err)
-
-		return nil, err
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", ns, name, err)
 	}
 
-	for k, v := range secret.Data {
-		deviceSecret[k] = string(v)
+	for k, v := range secretData {
+		deviceSecret[k] = v
 	}
 
 	return deviceSecret, nil
@@ -111,7 +109,7 @@ func callNodeStageVolume(ns *NodeServer, c *k8s.Clientset, pv *v1.PersistentVolu
 	log.DefaultLog("sending nodeStageVolume for volID: %s, stagingPath: %s",
 		volID, stagingParentPath)
 
-	deviceSecret, err := getSecret(c,
+	deviceSecret, err := getSecret(
 		pv.Spec.PersistentVolumeSource.CSI.NodeStageSecretRef.Namespace,
 		pv.Spec.PersistentVolumeSource.CSI.NodeStageSecretRef.Name)
 	if err != nil {

